great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -532,13 +547,17 @@ def create_temporary_table(self, table_name, custom_sql):
         It hasn't been tested in all SQL dialects, and may change based on community feedback.
         :param custom_sql:
         """
-        if self.engine.dialect.name == "mysql":
+
+        if self.engine.dialect.name.lower() == "bigquery":
+            stmt = "CREATE OR REPLACE TABLE `{table_name}` AS {custom_sql}".format(
```

Comment:
Unfortunately, BQ has very limited support for temporary tables. They just released a beta for creating temp tables with scripting, but according this [thread](https://stackoverflow.com/questions/20673986/how-to-create-temporary-table-in-google-bigquery) with BQ maintainers, the temp table lives only as long as the script, so it will not work for us here.

The way it's currently written, the user must pass a batch kwarg `bigquery_temp_table` that fully specifies the project, dataset, and table they want to use as a temp, otherwise the query will fail ungracefully. Would alerting them with a more graceful error and a note in the BQ documentation suffice?